### PR TITLE
chore(#75): Add tag-and-release ci job

### LINKED ISSUE #75: Configure versioning for application repo
*Do not assign a story-point value to this task*

### What needs to be done
Configure automatic versioning based on commit messages

### Out of scope
Publishing assets, only configure tagging on the repo

### PR DESCRIPTION
**Pull request by alexstojda**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Add versioning to the project

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos if the pull request changes UI.

### Closing issues
List all issues this PR resolves.
- closes #75 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated (n/a)
- [ ] All commits follow naming conventions

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,6 +8,38 @@ orbs:
   sonarcloud: sonarsource/sonarcloud@1.0.2
   slack: circleci/slack@3.4.2
 jobs:
+  tag-and-release:
+    docker:
+      - image: cimg/base:2020.01
+    environment:
+      TAG_FORMAT: "{Major}.{Minor}.{Patch}"
+      GITHUB_OWNER: csu-booking-platform
+      GITHUB_REPO: application
```
Comment (EvanDime) [inline .circleci/config.yml:17]: Wouldn't the owner be the CSU and the repo be the CSU-booking-platform?

**Review by EvanDime — APPROVED** [review]
Besides my comment, this looks good.

File: .circleci/config.yml (hunk 1 above)
Comment (alexstojda) [reply .circleci/config.yml:17]: Its the github organization name & repo name. 

The https://github.com/csu-booking-platform/application belongs to the org we created for capstone: https://github.com/csu-booking-platform

**Review by jonathanp16 — APPROVED** [review]
Looks good.

**Pull request merged**